src/components/SlidePanel/SlidePanel.js

```diff
@@ -1,8 +1,30 @@
-import React from "react";
+import React, { useEffect } from "react";
 
 import "./_slide-panel.scss";
 
 function SlidePanel({ children, onClose, isActive }) {
+  // If Escape key is pressed when slide panel is open, close it
+  useEffect(() => {
```

Comment:
This will still attach the events if the panel is open or not, maybe instead you could move the `isActive` conditional around the body of the `useEffect` function?